src/misc/helper.ts

```diff
@@ -0,0 +1,142 @@
+import { writeFileSync, statSync, existsSync, mkdirSync, readdirSync, appendFileSync } from 'fs';
+import {
+  VERSION,
+  DELIMITER,
+  BLOCKCHAIN_DIR,
+  JITCOIN_DIR,
+  JITCOIN_FILE,
+  MAX_FILE_SIZE,
+  JITCOIN_FILE_ZEROS,
+} from './constants';
+import { stringify } from 'querystring';
+import { Transaction } from '../jitcoin/block';
+import { BlockHeader, TransactionElement, BlockBody } from './interfaces';
+
+/**
+ *
+ * @author Flo Dörr
+ * @param {string} previousBlockHash
+ * @param {string} merkleTree
+ * @param {number} nonce
+ * @param {Transaction[]} transactions
+ */
+export const saveBinaryHex = (
+  previousBlockHash: string,
+  merkleTree: string,
+  nonce: number,
+  transactions: Transaction[],
+) => {
+
+  createDir();
+
+  const file = getJitCoinFile();
+
+  // write magic bytes
+  const delimiterHex = Buffer.from(DELIMITER).toString('hex');
+
+  // write header
+  const header = {
+    version: VERSION,
+    previousBlockHash,
+    merkleTree,
+    nonce,
+    time: new Date().getTime(),
+  } as BlockHeader;
+
+  // write header
+  const headerHex = Buffer.from(stringify(header)).toString('hex');
+
+  // write transaction count
+  const lengthHex = Buffer.from(transactions.length.toString()).toString(
+    'hex',
+  );
+
+  const trans = new Array<string>();
+
+  for (const transaction of transactions) {
+    trans.push(
+      Buffer.from(stringify({
+        amount: transaction.amount,
+        randomHash: transaction.randomHash,
+        userId: transaction.userId,
+      } as TransactionElement)).toString('utf8'),
+    );
+  }
+
+  // write body
+  const body = {
+    transactions: trans,
+  } as BlockBody;
+
+  // write body
+  const bodyHex = Buffer.from(stringify(body)).toString('hex');
+
+  const sizeHex = Buffer.from(
+    '' + (headerHex.length + lengthHex.length + bodyHex.length),
+  ).toString('hex');
+
+  const data = delimiterHex + sizeHex + headerHex + lengthHex + bodyHex;
+
+  appendFileSync(file, data, { encoding: 'hex' });
+};
+
+/**
+ *
+ * @author Flo Dörr
+ * @param {string} filename
+ * @returns {number} filesize in bytes
+ */
+const getFilesize = (filename: string): number => {
+  const stats = statSync(filename);
+  const fileSizeInBytes = stats.size;
+  return fileSizeInBytes;
+};
+
+/**
+ *
+ * @author Flo Dörr
+ */
+const createDir = () => {
+  if(!existsSync(JITCOIN_DIR)){
+    mkdirSync(JITCOIN_DIR);
+  }
+  if(!existsSync(BLOCKCHAIN_DIR)){
+    mkdirSync(BLOCKCHAIN_DIR);
+  }
+};
+
+/**
+ *
+ * @author Flo Dörr
+ * @returns {string} current file
+ */
+const getJitCoinFile = (): string => {
+  const files = readdirSync(BLOCKCHAIN_DIR);
```

Comment:
Are you sure it's a good idea to block the thread and load the files synchronously? I'd rather use the built-in streaming wrapped with some Promises 👍

Reply:
I wanted to add Promises in a later Ticket because there a yet other occasions where async would be nice (i.e. the mining).